packages/datepicker/src/ux-datepicker.ts

```diff
@@ -178,12 +215,34 @@ export class UxDatepicker implements UxComponent {
     this.styleEngine.applyTheme(newValue, this.element);
   }
 
-  private processAttribute(attributeName: string) {
-    const attributeValue = this.element.getAttribute('placeholder');
+  public focusedChanged(focused: boolean) {
+    this.element.dispatchEvent(DOM.createCustomEvent(focused ? 'focus' : 'blur', { bubbles: false }));
+  }
 
-    if (attributeValue) {
-      this.element.removeAttribute(attributeName);
-      this.textbox.setAttribute(attributeName, attributeValue)
+  public focusInput() {
+    this.textbox.focus();
+  }
+
+  public variantChanged(newValue: string) {
+    if (newValue === 'outline') {
+      let parentBackgroundColor = '';
```

Comment:
It feels like this could be extracted into a common utility in core package as there's probably 4 or 5 ux components that use this?

Reply:
Absolutely !